zilliqa/src/consensus.rs

```diff
@@ -2219,14 +2274,21 @@ impl Consensus {
         let hash = block.hash();
         debug!(?from, ?hash, ?block.header.view, ?block.header.number, "added block");
         let _ = self.new_blocks.send(block.header);
-        if let Some(child_proposal) = self.block_store.process_block(from, block)? {
-            self.message_sender.send_external_message(
-                self.peer_id(),
-                ExternalMessage::BlockResponse(BlockResponse {
-                    proposals: vec![child_proposal],
-                }),
-            )?;
-        }
+        // We may have child blocks; process them too.
+        self.block_store
+            .process_block(from, block)?
+            .into_iter()
+            .for_each(|(from_id, child_proposal)| {
+                // If we fail here, just fail - eventually someone will notice we did and
+                // retry the whole action.
```

Comment:
OK - that's a fair comment; originally, because if you fail here for one block you will throw all the others away and if that failure was because a buffer was full or similar, you might well end up in a repeating loop. Just let's remember not to make send_external_message() fail for another reason!